### PR TITLE
Explain the module specific themes separation capability of Silverstripe

### PR DESCRIPTION
**Pull request by jedateach**
A problem I had recently highlighted the need for some missing docs on how the `theme_module` setup is possible.

 #3944

### CODE REVIEW

**Comment by dhensby** [comment]
:mind_blown: is this true?! Are there even any tests for this behaviour?

**Comment by jedateach** [comment]
Ya - its been around since 2.3 / 2.4

I think this line is the best proof:
https://github.com/silverstripe/silverstripe-framework/blob/3.1/core/manifest/TemplateManifest.php#L189

Not sure if there are tests.

File: docs/en/02_Developer_Guides/01_Templates/05_Template_Inheritance.md [hunk 1]
```diff
@@ -46,9 +46,10 @@ your entire project for the appropriate `.ss` files located in `template` direct
 It will each and prioritize templates in the following priority:
 
 1. mysite (or other name given to site folder)
-2. themes
-3. modules 
-4. framework.
+2. module-specific themes (e.g. themes/simple_blog)
```
Comment (micmania1) [inline docs/en/02_Developer_Guides/01_Templates/05_Template_Inheritance.md]: Is this definitely the right priority? I answered this a while back: http://stackoverflow.com/questions/24969798/blog-template-isnt-overridden - I'm not sure if I'm right but I'm guessing I would have tested at the time. I'm fairly certain that themes are higher priority then mysite.

Comment (dhensby) [inline docs/en/02_Developer_Guides/01_Templates/05_Template_Inheritance.md]: My understanding is that mysite trumps all

Comment (micmania1) [inline docs/en/02_Developer_Guides/01_Templates/05_Template_Inheritance.md]: Appears so :)

**Pull request merged**